src/integrationTest/java/com/mongodb/hibernate/EmbeddableIntegrationTests.java

```diff
@@ -0,0 +1,413 @@
+            EmbeddableIntegrationTests.Unsupported.ItemWithNestedValueHavingNonUpdatable.class
+        })
+@ExtendWith(MongoExtension.class)
+class EmbeddableIntegrationTests implements SessionFactoryScopeAware {
+    @InjectMongoCollection("items")
+    private static MongoCollection<BsonDocument> mongoCollection;
+
+    private SessionFactoryScope sessionFactoryScope;
+
+    @Test
+    void testFlattenedValues() {
+        var item = new ItemWithFlattenedValues();
+        {
+            item.id = new EmbeddableValue();
+            item.id.a = 1;
+            item.flattened1 = new EmbeddableValue();
+            item.flattened1.a = 2;
+            item.flattened2 = new EmbeddablePairValue1();
+            item.flattened2.a = 3;
+            item.flattened2.flattened = new EmbeddablePairValue2(4, 5);
+            item.flattened2.parent = item;
+        }
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id.a))
+                        .append("flattened1_a", new BsonInt32(2))
+                        .append("flattened2_a", new BsonInt32(3))
+                        .append("flattened2_flattened_a", new BsonInt32(4))
+                        .append("flattened2_flattened_b", new BsonInt32(5)));
+        var loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithFlattenedValues.class, item.id));
+        assertEquals(item, loadedItem);
+        var updatedItem = sessionFactoryScope.fromTransaction(session -> {
+            var result = session.find(ItemWithFlattenedValues.class, item.id);
+            result.flattened1.a = -result.flattened1.a;
+            return result;
+        });
+        loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithFlattenedValues.class, item.id));
+        assertEquals(updatedItem, loadedItem);
+    }
+
+    @Test
+    void testFlattenedEmptyValue() {
+        var item = new ItemWithFattenedEmptyValue();
+        {
+            item.id = 1;
+            item.omitted = new EmbeddableEmptyValue();
+        }
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
+                .containsExactly(
+                        // Hibernate ORM does not store/read the empty `item.omitted` value.
+                        // See https://hibernate.atlassian.net/browse/HHH-11936 for more details.
+                        new BsonDocument(ID_FIELD_NAME, new BsonInt32(item.id)));
+        var loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithFattenedEmptyValue.class, item.id));
+        // the entity we stored and the entity we loaded are not equal because Hibernate ORM omits `item.omitted`
+        assertNotEquals(item, loadedItem);
```

Comment:
The current usage of `aspectj` is fine, though not idematic.
If only one field is the blocker to equalness, usually assertj's `ignoringFields()` method is used (see https://www.baeldung.com/assertj-ignore-fields-comparison). That way, we convey the two assertion intentions in one go:

1. one field is different
2. all the other fields are equals

which is a little bit better than `assertNotEquals`. Furthermore, we don't need to explicitly set the field to null simply to be able to run `assertEquals`.

Optional; only for discussion purpose.

Reply:
Thank you. I changed this in e138a818c97ad390a75e2b77c4f2889935ff6950.